src/main/java/seedu/address/model/TodoList.java

```diff
@@ -0,0 +1,144 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.task.Task;
+import seedu.address.model.task.UniqueTodoList;
+import seedu.address.model.task.comparator.TaskComparatorByDate;
+import seedu.address.model.task.comparator.TaskComparatorByName;
+import seedu.address.model.task.comparator.TaskComparatorByPriority;
+
+/**
+ * Wraps all data at the todo list level.
+ * Duplicates are not allowed (by .isSameTask comparison)
+ */
+public class TodoList implements ReadOnlyTodoList {
+
+    private final UniqueTodoList tasks;
+
+    /*
+     * The 'unusual' code block below is a non-static initialization block, sometimes used to avoid duplication
+     * between constructors. See https://docs.oracle.com/javase/tutorial/java/javaOO/initial.html
+     *
+     * Note that non-static init blocks are not recommended to use. There are other ways to avoid duplication
+     *   among constructors.
+     */
+    {
+        tasks = new UniqueTodoList();
+    }
+
+    public TodoList() {}
```

Comment:
I am using the same implementation as our current ModuleList class (from AB-3). The reason is because module list is very similar to todo list. I think it is a good idea to maintain consistency and similarity for the implementation throughout the code? 

Or perhaps we should instead try to implement our own version of todo list that does replicate the implementation of module list? i.e. todo list in cap5buddy package uses a different implementation by using arrayList instead.